src/lib/chawathe98_matcher.rs

```diff
@@ -46,35 +46,2262 @@
+                        // //// CASE 1.3 //// //
+                        else if list_m[0].id() == self.from_arena.borrow().size() - 1
+                            && list_n[0].id() == self.to_arena.borrow().size() - 1
+                            {
+                                let mut edge = Edge::new(list_m[0], list_n[0], 1, EdgeType::NULL);
+                                list_edges.push(edge);
+                            }
+                            // //// CASE 1.4 //// //
+                            else if list_m[0].id() == m.id() && m.id() != self.from_arena.borrow().size() - 1
+                                && list_n[0].id() == n.id()
+                                && n.id() != self.to_arena.borrow().size() - 1
+                                {
+                                    // Check if parent of m,n is in edges
+                                    for check in edges.clone() {
+                                        let m_parent = self.from_arena.borrow()[m].parent().unwrap();
+                                        let n_parent = self.to_arena.borrow()[n].parent().unwrap();
+                                        if m_parent.id() == check.from_node.id()
+                                            && n_parent.id() == check.to_node.id()
+                                            {
+                                                let mut edge = Edge::new(m, n, 1, EdgeType::NULL);
+                                                list_edges.push(edge);
+                                            } else {
+                                            let mut edge = Edge::new(m, n, 1, EdgeType::MOVE);
+                                            list_edges.push(edge);
+                                        }
+                                    }
+                                }
+            }
+                // ///// CASE 2 //// //
+                else if list_m.len() == 1 && list_n.len() > 1 {
+                    // //// CASE 2.1 //// //
+                    // Figure 16, appendix D
+                    if list_m[0].id() == m.id() && m.id() != self.from_arena.borrow().size() - 1 {
+                        // Partition
+                        for n1 in &list_n {
+                            for n2 in &list_n {
+                                // Check if n1.parent == n2.parent are the same
+                                if n1.id() != n2.id() {
+                                    //                                    let n1_parent = self.from_arena[m].parent().unwrap();
+                                    //                                    let n2_parent = self.to_arena[n].parent().unwrap();
+                                }
+                            }
+                        }
+                    }
+                        // //// CASE 2.2 //// //
+                        // Figure 17, appendix D
+                        else if list_m[0].id() == self.from_arena.borrow().size() - 1 {
+
```

Comment:
It looks like this `if` statement and the one above don't do anything? i.e. there's no code inside the braces?